File: version_java/src/main/java/com/example/version_java/data/repo/KakaoRepositoryImpl.java
package com.example.version_java.data.repo;

import com.example.version_java.data.source.remote.KakaoRemoteDataSource;

import javax.inject.Inject;

import kotlin.jvm.functions.Function1;

public class KakaoRepositoryImpl implements KakaoRepository {

    private final KakaoRemoteDataSource kakaoRemoteDataSource;

    @Override
    public void getSearchList(String location, Function1 callback) {
        kakaoRemoteDataSource.getSearchList(location, callback);
    }

    @Inject
    KakaoRepositoryImpl(
            KakaoRemoteDataSource kakaoRemoteDataSource
    ) {
        this.kakaoRemoteDataSource = kakaoRemoteDataSource;
    }
}

File: app/src/main/java/com/example/crimenotification/data/repo/KakaoRepositoryImpl.kt
package com.example.crimenotification.data.repo

import com.example.crimenotification.data.source.remote.KakaoRemoteDataSource
import com.example.crimenotification.network.response.KakaoSearchResponse
import com.example.crimenotification.util.Result
import javax.inject.Inject

class KakaoRepositoryImpl @Inject constructor(private val kakaoRemoteDataSource: KakaoRemoteDataSource) :
    KakaoRepository {
    override fun getSearchList(
        location: String,
        callback: (result: Result<KakaoSearchResponse>) -> Unit
    ) {
        kakaoRemoteDataSource.getSearchList(location, callback)
    }
}